clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -827,6 +827,165 @@ CodeGenFunction::evaluateOrEmitBuiltinObjectSize(const Expr *E, unsigned Type,
   return ConstantInt::get(ResType, ObjectSize, /*isSigned=*/true);
 }
 
+llvm::Value *
+CodeGenFunction::emitFlexibleArrayMemberSize(const Expr *E, unsigned Type,
+                                             llvm::IntegerType *ResType) {
```

Comment:
it's hard to tell what changed since this patch:
1. creates `CodeGenFunction::emitFlexibleArrayMemberSize`
2. changes logic within `CodeGenFunction::emitFlexibleArrayMemberSize`

Mind pre-committing 1 so that it's easier to see what logic was changed in isolation?

Reply:
I'll create a NFC patch that just does the refactoring and rebase this onto it.

Reply:
This has been rebased with the NFC refactor. PTAL.